lib/weatherApp.js

```diff
@@ -0,0 +1,46 @@
+(function (factory) {
+    if (typeof define === 'function' && define['amd']) {
+        define(['knockout'], factory);
+    } else if (typeof require === 'function' && typeof exports === 'object' && typeof module === 'object') {
+        module.exports = factory(require('knockout'));
+    } else {
+        window.WeatherApp = factory(ko);
+    }
+} (function (ko) {
+    return function WeatherApp() {
+        'use strict';
+        let city = ko.observable("");
+        let api = "http://api.openweathermap.org/data/2.5/weather?q=";
+        let unit = "&units=metric&";
+        let apiId = "appid=bd959553bce6759da749a8bcad48f038";
+        let maxTemp = ko.observable("");
+        let minTemp = ko.observable("");
+        let isTempVisible = ko.observable(false);
+        let josnCityName = ko.observable("");
+
+        function showWeather() {
+            let url = api + city() + unit + apiId;
+            fetch(url)
+                .then((response => {
+                    response.json()
+                        .then(josn => {
+                            maxTemp(josn.main.temp_max);
+                            minTemp(josn.main.temp_min);
+                            isTempVisible(true);
+                            josnCityName(josn.name);
```

Comment:
It would be good to include the country from the results as well. For example if a user types in "York" and presses submit, they will get the weather for York in UK and it will display "York". But what if I wanted to know the weather for "York" in the US? If I typed in "York, US", it will return the correct weather, but the city name will still just say "York". So there's no way for the user to know if the York that was returned was the one they were expecting or not.